src/snowflake/snowpark/_internal/code_generation.py

```diff
@@ -0,0 +1,594 @@
+    """
+    func_module_name = func.__module__
+    func_text = ""
+    global_vars_text = ""
+    for name, obj in ref_objects.items():
+        if obj == func:
+            continue
+        if inspect.ismodule(obj):
+            # a) imported modules
+            to_import.add((obj.__name__, name))  # name could be an alias
+        elif (
+            inspect.isclass(obj) or inspect.isfunction(obj)
+        ) and obj.__module__ != func_module_name:
+            # b) classes or functions imported from other modules
+            to_import_from_module[obj.__module__].add(
+                (obj.__name__, name)
+            )  # name could be an alias
+        else:
+            # function/class/variables defined in the same module
+            if inspect.isfunction(obj):
+                func_text = f"{func_text}{textwrap.dedent(inspect.getsource(obj))}"
+            elif inspect.isclass(obj):
+                # dynamic class parsing will be handled separately
+                pass
+            else:
+                # c) global variables used by UDF
+                if code_as_comment:
+                    global_vars_text = (
+                        f"{global_vars_text}{name}  # variable of type {type(obj)}\n"
+                    )
+                    continue  # skip the serialization part if we just need code as comment
+                # v does not have source code, then it's a global variable of which the value has been evaluated
+                try:
+                    global_vars_text = f"""\
+{global_vars_text}
+{name} = pickle.loads(bytes.fromhex('{pickle.dumps(obj).hex()}'))  \
+# {name} is of type {type(obj)} and serialized by snowpark-python
+"""
+                except Exception as exc:
+                    logger.debug(
+                        f"Unable to generate source code for object {name} of type {type(obj)} due to exception {exc}"
+                    )
+                    raise
+    return func_text, global_vars_text
+
+
+def resolve_target_func_imports(
+    to_import: Set[Tuple[str, str]],
+    to_import_from_module: Dict[str, Set[Tuple[str, str]]],
+) -> str:
+    """
+    Deal with imports and alias, generate imports string.
+
+    Args:
+        to_import: set of name and alias pairs of direct imports which should be generated as "import xxx"
+            or "import xxx as yyy".
+        to_import_from_module: dict of import information, key is the module name with value being the set of
+            name and alias paris of imported objects which should be generated as "from xxx import yyy" or
+            "from xxx import yyy as zzz".
+
+    Returns:
+        A string of generated imports.
+    """
+    imports = [
+        f"import {name + ' as ' if name != alias else ''}{alias}"
+        for name, alias in sorted(to_import)
+    ]
+    for module, name_alias_pairs in sorted(to_import_from_module.items()):
+        classes = ", ".join(
+            f"{name + ' as ' if name != alias else ''}{alias}"
```

Comment:
If we explicitly check for `name != alias`, then we might as well just set `alias` to `None`, and write a small helper function to generate this?

Reply:
I have introduced a named tuple for the ImportNameAliasPair and defaults both name and alias None.